pkg/collector/corechecks/snmp/internal/profile/testing_utils.go

```diff
@@ -10,33 +10,16 @@ import (
 	"path/filepath"
 	"regexp"
 
+	"github.com/mohae/deepcopy"
+
 	"github.com/DataDog/datadog-agent/pkg/config"
 
 	"github.com/DataDog/datadog-agent/pkg/networkdevice/profile/profiledefinition"
 )
 
 // CopyProfileDefinition copies a profile, it's used for testing
-// TODO: Use deepcopy library instead?
 func CopyProfileDefinition(profileDef profiledefinition.ProfileDefinition) profiledefinition.ProfileDefinition {
-	newDef := profiledefinition.ProfileDefinition{}
-	newDef.Metrics = append(newDef.Metrics, profileDef.Metrics...)
-	newDef.MetricTags = append(newDef.MetricTags, profileDef.MetricTags...)
-	newDef.StaticTags = append(newDef.StaticTags, profileDef.StaticTags...)
-	newDef.Metadata = make(profiledefinition.MetadataConfig)
-	newDef.Device = profileDef.Device
-	newDef.Extends = append(newDef.Extends, profileDef.Extends...)
-	newDef.SysObjectIds = append(newDef.SysObjectIds, profileDef.SysObjectIds...)
-
-	for resName, resource := range profileDef.Metadata {
-		resConfig := profiledefinition.MetadataResourceConfig{}
-		resConfig.Fields = make(map[string]profiledefinition.MetadataField)
-		for fieldName, field := range resource.Fields {
-			resConfig.Fields[fieldName] = field
-		}
-		resConfig.IDTags = append(resConfig.IDTags, resource.IDTags...)
-		newDef.Metadata[resName] = resConfig
-	}
-	return newDef
+	return deepcopy.Copy(profileDef).(profiledefinition.ProfileDefinition)
```

Comment:
Replace previous copy logic with Deepcopy. The previous copy code was always providing a non-nil metadata `newDef.Metadata = make(profiledefinition.MetadataConfig)` that was "hiding" the bug in the tests.

This change makes the test fail if the fix in profile_resolver.go is not present.